client/asset/eth/eth.go

```diff
@@ -3740,7 +3769,11 @@ func (w *assetWallet) redeem(ctx context.Context, assetID uint32, redemptions []
 
 	return tx, w.withContractor(contractVer, func(c contractor) error {
 		tx, err = c.redeem(txOpts, redemptions)
-		return err
+		if err != nil {
+			c.voidUnusedNonce()
```

Comment:
The `redeem` method is where I was describing the `nonceOverride` issue (see above).
Problems:
- It calls `w.node.txOpts`, getting (and reserving) a new nonce, and then it uses the `nonceOverride`.  If `nonceOverride` was set, we should either (a) immediately void/return the nonce when overriding `txOpts.Nonce` above, or (b) modify `txOpts()` with an input arg containing the nonce to use instead of getting a new one, or perhaps just a bool arg indicating not to set the nonce in the txopts at all.
- In the `err` condition here after `c.redeem`, we should not return the nonce if we've used the override and thus not reserved one.

Reply:
Oh! I finally understand. For redeem we may need to use a previously used nonce. Fixing...

Reply:
> (b) modify txOpts() with an input arg containing the nonce to use instead of getting a new one

Will do this.